gammapy/datasets/tests/test_map.py

```diff
@@ -262,6 +262,20 @@ def test_map_dataset_str(sky_model, geom, geom_etrue):
     assert "MapDataset" in str(dataset)
 
 
+def test_map_dataset_to_asimov(sky_model, geom, geom_etrue):
+    dataset = get_map_dataset(geom, geom_etrue)
+
+    bkg_model = FoVBackgroundModel(dataset_name=dataset.name)
+    dataset.models = [sky_model, bkg_model]
+
+    npred_sum = dataset.npred().data.sum()
+
+    asimov_dataset = dataset.to_asimov_dataset()
+    counts_asimov = asimov_dataset.counts.data.sum()
```

Comment:
maybe also assert that the model is correctly reassigned

Reply:
done